purchase_operating_unit_sequence/models/purchase_order.py

```diff
@@ -0,0 +1,13 @@
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.model
+    def create(self, vals):
+        if vals.get("operating_unit_id", False):
```

Comment:
I guess we should use
`if vals.get("name", "/") == "/" and vals.get("operating_unit_id", False):`

Reply:
In purchase.order we don't have vals.get("name") it's not set at all like in sale.order for example